### PR TITLE
migrations, entity preparation and flow control (Task item 1)

### PR DESCRIPTION
**Pull request by romanetar**
ref https://tipit.avaza.com/project/view#!tab=task-pane&groupby=MyTaskDueDate&view=vertical&task=2824384&fileview=grid

Tix Point 1 completed and ready to review
Tix  Point 2 WIP

Signed-off-by: romanetar <roman_ag@hotmail.com>

### CODE REVIEW

File: app/Models/Foundation/Summit/Summit.php [hunk 1]
```diff
@@ -431,6 +431,12 @@ class Summit extends SilverstripeBaseModel
      */
     private $allow_update_attendee_extra_questions;
 
+    /**
+     * @ORM\Column(name="AllowAutomaticReminderEmails", type="boolean")
```
Comment (smarcet) [inline app/Models/Foundation/Summit/Summit.php:435]: please rename db field to **RegistrationAllowAutomaticReminderEmails**

File: database/migrations/model/Version20220211202106.php [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+{
+    /**
+     * @param Schema $schema
+     */
+    public function up(Schema $schema): void
+    {
+        $builder = new Builder($schema);
+        if($schema->hasTable("Summit") && !$builder->hasColumn("Summit", "AllowAutomaticReminderEmails ")) {
+            $builder->table('Summit', function (Table $table) {
+                $table->boolean('AllowAutomaticReminderEmails ')->setNotnull(true)->setDefault(true);
```
Comment (smarcet) [inline database/migrations/model/Version20220211202106.php:35]: please rename db field to **RegistrationAllowAutomaticReminderEmails**

**Comment by smarcet** [comment]
@romanetar its missing the update at summit serializer 
https://github.com/OpenStackweb/summit-api/blob/main/app/ModelSerializers/Summit/SummitSerializer.php
and the update to summit validation rules factory
https://github.com/OpenStackweb/summit-api/blob/main/app/Http/Controllers/Apis/Protected/Summit/Factories/SummitValidationRulesFactory.php
also review inline comments
regards

**Review by smarcet — CHANGES_REQUESTED** [review]
please review comments

File: database/migrations/model/Version20220211202106.php (hunk 2 above)
Comment (romanetar) [reply database/migrations/model/Version20220211202106.php]: Done

File: app/Models/Foundation/Summit/Summit.php (hunk 1 above)
Comment (romanetar) [reply app/Models/Foundation/Summit/Summit.php]: Done

File: app/Models/Foundation/Summit/Summit.php [hunk 3]
```diff
@@ -6087,6 +6095,22 @@ public function setAllowUpdateAttendeeExtraQuestions(bool $allow_update_attendee
         $this->allow_update_attendee_extra_questions = $allow_update_attendee_extra_questions;
     }
 
+    /**
+     * @return bool
+     */
+    public function allowsAutomaticReminderEmails(): bool
```
[the remark this answers is not in the document]
Comment (romanetar) [reply app/Models/Foundation/Summit/Summit.php]: Prefixed with is

**Review by smarcet — APPROVED** [review]
looks good to me

**Pull request merged**